### DISCUSSION TITLE
Is there a way to proxy a query to another subgraph?

### DISCUSSION THREAD

**Discussion by Rfank2021** [Netflix/dgs-framework#1168] [Q&A]
We want to have an api in our subgraph returning same data schema as another team's subgraph api, so the query schema is same too. Only difference is the input variables and validations, how to implement a query proxy?

**Comment by srinivasankavitha** [comment]
I don't think there is a way to do so with the existing federation mechanism. The other team's api will have to delegate the calls to your service based on the input variables in that case. 

You will need to have a proxy service that fulfills the api. The gateway would talk to this proxy service and the proxy service would then determine which subgraph to call.

**Reply by Rfank2021** [reply]
The other team's api
```
type Query {
  user: UserInfo
}
```

Our api
```
type Query {
  specialUser(encryptedUserId: ID): UserInfo
}
```
In order to use the same query/return schema, we need `@shareable` which need redefine all fields, not so good. I was trying to extract query and send the query to the other team's api, but didn't find a way in backend.

**Comment by srinivasankavitha** [comment]
To be clear, are you trying to just use the UserInfo type from the other
team? And my assumption is that your services are behind a federated
gateway? (since you are referring to subgraphs).

If you are just looking to use the other team's type in your schema, you
will need to redefine the type locally to be able to use it in your schema.

**Reply by Rfank2021** [reply]
Yes, all teams' services are under same gateway.

**Reply by Rfank2021** [reply]
Even redefine it locally, I still didn't find way to forward the query.

**Reply by srinivasankavitha** [reply]
Can you explain more about what you mean by forward the query? I assume you just need to send another GraphQL query directly to the other service from your service.

Perhaps it would be clearer if you can list out the steps more explicitly. From the schema you shared, it looks like you are just using the `UserInfo` type in your service? Or do you need to get data from the other service?

**Reply by Rfank2021** [reply]
This is an example in nodejs before federation, https://dev.to/harshitkumar31/creating-a-proxy-for-your-graphql-server-5b9o

**Reply by Rfank2021** [reply]
`Or do you need to get data from the other service?`, of course since our subgraph can't resolve `UserInfo`.
